### PR TITLE
Migrate to TinyXML2

### PR DESCRIPTION
**Pull request by MatthijsBurgh**

### CODE REVIEW

File: src/models/xml_shape_parser.cpp [hunk 1]
```diff
@@ -35,18 +36,18 @@ geo::ShapePtr parseXMLShape(const std::string& filename, std::string& error)
...
-    TiXmlDocument doc(filename);
-    doc.LoadFile();
+    tinyxml2::XMLDocument doc;
+    doc.LoadFile(filename.c_str());
 
     if (doc.Error())
     {
         s_error << "While parsing '" << filename << "': " << std::endl << std::endl
-                << doc.ErrorDesc() << " at line " << doc.ErrorRow() << ", col " << doc.ErrorCol() << std::endl;
+                << doc.ErrorStr() << " at line " << doc.ErrorLineNum() << std::endl;
```
Comment (jlunenburg) [inline src/models/xml_shape_parser.cpp:45]: No column number?

File: src/models/xml_shape_parser.cpp [hunk 2]
```diff
@@ -35,18 +36,18 @@ geo::ShapePtr parseXMLShape(const std::string& filename, std::string& error)
...
         error = s_error.str();
         return geo::ShapePtr();
     }
 
-    const TiXmlElement* model_xml = doc.FirstChildElement("model");
+    const tinyxml2::XMLElement* model_xml = doc.FirstChildElement("model");
```
Comment (jlunenburg) [inline src/models/xml_shape_parser.cpp:50]: Could we used `unique_ptr`?

**Review by jlunenburg — APPROVED** [review]
Assume it's been tested

File: src/models/xml_shape_parser.cpp (hunk 1 above)
Comment (MatthijsBurgh) [reply src/models/xml_shape_parser.cpp:45]: No, that feature has been removed.

File: src/models/xml_shape_parser.cpp (hunk 2 above)
Comment (MatthijsBurgh) [reply src/models/xml_shape_parser.cpp:50]: You could probably get it working. But I don't think it is faster. I prefer to keep it this way for now.

**Pull request merged**